File: Backend/app/Models/Purchase.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Support\Facades\DB;
use Illuminate\Database\Eloquent\SoftDeletes;

class Purchase extends Model
{
    use HasFactory;
    use SoftDeletes;

    protected $fillable = [
        'date_time',
        'total_pay',
        'status',
        'paid',
        'user_id',
        'desk_id',
        'product_id'
    ];
    public $timestamps = false;
    protected $dates = ['deleted_at'];

    public function desk():BelongsTo
    {
        return $this->belongsTo(Desk::class);
    }

    public function user():BelongsTo
    {
        return $this->belongsTo(User::class)->whereNotNull('id');
    }

    public function products():BelongsToMany
    {
        return $this->belongsToMany(Product::class, 'product_purchase')->as('product_purchase')->withPivot('quantity');
    }

    public function recalculateTotalPay()
    {
        $totalPay = $this->products()->sum(DB::raw('price * quantity'));

        $this->total_pay = $totalPay;
        $this->save();
    }
}

File: MG_Admin_GUI_v2.2/Models/purchase.cs
using System;
using System.Collections.Generic;

namespace MG_Admin_GUI.Models;

public partial class purchase
{
    public ulong id { get; set; }

    public DateTime date_time { get; set; }

    public int total_pay { get; set; }

    public string status { get; set; } = null!;

    public bool paid { get; set; }

    public ulong? user_id { get; set; }

    public ulong desk_id { get; set; }

    public string stripe_id { get; set; } = null!;

    public DateTime? deleted_at { get; set; }

    public virtual desk desk { get; set; } = null!;

    public virtual ICollection<product_purchase> product_purchases { get; set; } = new List<product_purchase>();

    public virtual user? user { get; set; }
}
